### PR TITLE
Update data model

### PR DESCRIPTION
**Pull request by jackxujh**
This pull request updates the data model as described in #66, and includes:

- A new table for venues.
- A new table for sensor types.
- New and updated convenience methods in Simulator.
- Relevant tests.
- Move serialization methods into its dedicated branch.

### CODE REVIEW

File: ag_data/models.py [hunk 1]
```diff
@@ -13,15 +22,22 @@ class AGEvent(models.Model):
...
     event_description = models.CharField(max_length=100, null=False, blank=True)
+    event_venue = models.ForeignKey(AGVenue, null=True, on_delete=models.SET_NULL)
+
+
+class AGSensorType(models.Model):
+
+    sensorType_id = models.AutoField(primary_key=True)
+    sensorType_name = models.CharField(max_length=1024, blank=True)
+    sensorType_processingFormula = models.IntegerField(default=0, null=False)
+    sensorType_format = JSONField()
```
Comment (alldne) [inline ag_data/models.py]: So it's a mixed style of camelCase and snake_case. In any way, sticking to one style and being consistent would be important.  Personally I prefer use snake_case for variable names when I work on python [as pep8 suggests](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles).

File: ag_data/models.py [hunk 2]
```diff
@@ -4,6 +4,15 @@
 from django.utils import timezone
 
 
+class AGVenue(models.Model):
+
+    venue_uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    venue_name = models.CharField(max_length=100, blank=True)
+    venue_description = models.CharField(max_length=100, null=False, blank=True)
+    venue_latitude = models.FloatField()
+    venue_longitude = models.FloatField()
```
[the remark this answers is not in the document]
Comment (jackxujh) [reply ag_data/models.py]: Yes sure, will change to `DecimalField`.

File: ag_data/models.py (hunk 1 above)
Comment (jackxujh) [reply ag_data/models.py]: Good point! Will use snake_case.

Comment (jackxujh) [reply ag_data/models.py]: My original idea is to use the underscore to separate the table name and the field name. But it is not used now.

Comment (Ricardo-1894) [reply ag_data/models.py]: @alldne Good point, I am going to point that out on my on-call report

File: ag_data/models.py [hunk 3]
```diff
@@ -13,15 +22,22 @@ class AGEvent(models.Model):
     event_name = models.CharField(max_length=40, blank=True)
     event_date = models.DateTimeField(default=timezone.now)
     event_description = models.CharField(max_length=100, null=False, blank=True)
+    event_venue = models.ForeignKey(AGVenue, null=True, on_delete=models.SET_NULL)
```
[the remark this answers is not in the document]
Comment (jackxujh) [reply ag_data/models.py]: This is actually a very good point: no need to prefix field names with table name. I have changed this. Pushing the new version in...

**Pull request merged**